src/prefect/tasks/sftp/sftp.py

```diff
@@ -0,0 +1,280 @@
+        remote_path: str = None,
+        local_path: str = None,
+        **kwargs,
+    ):
+        self.host = host
+        self.username = username
+        self.password = password
+        self.port_number = port_number
+        self.remote_path = remote_path
+        self.local_path = local_path
+        super().__init__(**kwargs)
+
+    def _create_connection(self) -> None:
+        """
+        Initialise the connection with the SFTP server
+        :return: None
+        """
+        transport = Transport(sock=(self.host, self.port_number))
+        transport.connect(username=self.username, password=self.password)
+        self._connection = SFTPClient.from_transport(transport)
+        self.logger.info(f"connected to {self.host}, {self.port_number}")
+
+    @defaults_from_attrs(
+        "host", "username", "password", "port_number", "remote_path", "local_path"
+    )
+    def run(
+        self,
+        host: str = None,
+        username: str = None,
+        password: str = None,
+        port_number: int = None,
+        remote_path: str = None,
+        local_path: str = None,
+    ) -> bool:
+        """
+        Task for uploading files to an SFTP server.
+
+        Args:
+            - host (str): name of the host to use.
+            - username (str): username used to authenticate.
+            - password (str): password used to authenticate.
+            - port_number (int): the port number to connect to the server.
+            - remote_path (str): the remote sftp file path
+            - local_path (str): the local file path to from upload
+
+        Raises:
+            - ValueError: if a required parameter is not supplied.
+            - ClientError: if exception occurs when connecting/downloading from the server.
+        """
+        if not host:
+            raise ValueError("A host name must be provided")
+        if not username:
+            raise ValueError("User name must be provided")
+        if not password:
+            raise ValueError("A password must be provided")
+        if not port_number:
+            raise ValueError("A port_number name must be provided")
+        if not remote_path:
+            raise ValueError("A remote_path must be provided")
+        if not local_path:
+            raise ValueError("A local_path must be provided")
+
+        # first init connection to SFTP server
+        self._create_connection()
+
+        # download specified file/files
+        result = self.upload(self.local_path, self.remote_path)
+
+        # close sftp server connection
+        self._connection.close()
+
+        return result
+
+    def upload(self, local_path: str, remote_path: str) -> bool:
+        """
+        Uploads file to an SFTP remote server from the local path
+        to the specified remote path.
+        Args:
+            - local_path (str): the local file path to from upload
+            - remote_path (str): the remote sftp file path
+
+        Returns:
+            - bool: True/False
```

Comment:
Instead of returning a `Bool`, I suggest raising an exception if the upload fails.